client/files.go

```diff
@@ -364,3 +368,135 @@ func (client *Client) RemovePath(opts *RemovePathOptions) error {
 
 	return nil
 }
+
+// PushOptions contains the options for a call to Push.
+type PushOptions struct {
+	// Source is the source of data to write (required).
+	Source io.Reader
+
+	// Path indicates the absolute path of the file in the destination
+	// machine (required).
+	Path string
+
+	// MakeDirs, if true, will create any non-existing directories in the path
+	// to the remote file. If false, the default, the call to Push will
+	// fail if any non-existing directory is found on the remote path.
+	MakeDirs bool
+
+	// Permissions indicates the mode of the file in the destination machine.
+	// Defaults to 0644. Note that, when used together with MakeDirs, the
+	// directories that might be created will not use this mode, but 0755.
+	Permissions string
+
+	// UserID indicates the user ID of the owner for the file in the destination
+	// machine. When used together with MakeDirs, the directories that might be
+	// created will also be owned by this user.
```

Comment:
Suggestion (and similar below):

```suggestion
	// UserID indicates the user ID of the owner for the file on the destination
	// machine. When used together with MakeDirs, the directories that are
	// created will also be owned by this user.
```